.github/workflows/ci.yml

```diff
@@ -6,77 +6,76 @@ on:
-              </server>
-            </servers>
-          </settings>
-          EOF
+          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
+          echo "$major_version" | grep -q "^[0-9]\{1,\}$" || {
+            echo "ERROR: Failed to extract <major-version> from pom.xml"
+            exit 1
+          }
+          echo "major-version=$major_version" >> "$GITHUB_OUTPUT"
+
+      - uses: capralifecycle/actions-lib/generate-tag@v1
+        id: tag
+        with:
+          tag-prefix: "${{ steps.prep.outputs.major-version }}"
+          tag-type: "punctuated-timestamp-tag"
+
+      - name: resolve dependencies
+        env:
+          # NOTE: Use the following token if your project does not depend on private packages:
+          # GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # NOTE: Use the following token if your project depends on private packages:
+          GITHUB_TOKEN: ${{ secrets.SHARED_GITHUB_PACKAGES_TOKEN }}
+        run: mvn -B dependency:resolve
+
       - name: conditional release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          export MAVEN_OPTS="-Duser.home=$HOME" # Home directory not automatically picked up for some reason
-          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
-          date="$(date -u "+%Y%m%d.%H%M%S")"
-          tag="$major_version.$date"
-          printf "Tag to use for potential release: %s\n" "$tag"
-          mvn -B org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
+          SONAR_TOKEN: ${{ secrets.SHARED_SONAR_TOKEN }}
+          CONDITIONAL_RELEASE: ${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+          TAG: "${{ steps.tag.outputs.tag }}"
 
-          release="false"
-          git fetch --tags
-          if [ "$BRANCH_NAME" = "master" ]; then
-            latest_tag="$(git describe --abbrev=0 || true)"
-            latest_tag_commit="$(if test -n "$latest_tag"; then git rev-parse "$latest_tag^{}"; fi)"
-            if [ ! "$latest_tag_commit" = "$GITHUB_SHA" ]; then
-              release="true"
-            fi
-          fi
-          if [ "$release" = "true" ]; then
-            # Release to GitHub Packages
-            mvn -B source:jar deploy scm:tag -DaltReleaseDeploymentRepository="github::default::https://maven.pkg.github.com/capralifecycle/liflig-properties" -Drevision="$tag" -Dtag="$tag"
+        run: |
+          if [ "$CONDITIONAL_RELEASE" = "true" ]; then
+            echo "Releasing library with tag '$TAG'"
+            mvn -B source:jar deploy scm:tag -Drevision="$TAG" -Dtag="$TAG" \
+              org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
           else
-            mvn -B source:jar verify -Drevision="$tag" -Dtag="$tag"
+            mvn -B -U verify
```

Comment:
Do we want to run `source:jar` here as well?

Reply:
I don't `source:jar` will help provoke any more potential configuration issues, so considering we don't deploy the source/app jars from this stage I don't think we need it. Compile/test should be fine